applications/GeoMechanicsApplication/tests/cpp_tests/custom_utilities/test_extrapolation_utilities.cpp

```diff
@@ -0,0 +1,176 @@
+
+    Vector cauchy_stress(4);
+    cauchy_stress <<= 1000.0, 2000.0, 3000.0, 4000.0;
+    const auto          delta_stress = Vector(4, 1000.0);
+    std::vector<Vector> cauchy_stress_vectors;
+    cauchy_stress_vectors.push_back(cauchy_stress);
+    cauchy_stress_vectors.emplace_back(cauchy_stress + delta_stress);
+
+    std::vector<std::size_t> node_ids = {1, 2, 3};
+
+    // Act and Assert
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN((void)ExtrapolationUtilities::CalculateNodalStresses(
+                                          node_ids, element.GetGeometry(), element.GetIntegrationMethod(),
+                                          cauchy_stress_vectors, element.Id()),
+                                      "An extrapolation matrix size 3 is not equal to given "
+                                      "stress vectors size 2 for element Id 1");
+
+    cauchy_stress_vectors.emplace_back(cauchy_stress + 2 * delta_stress);
+
+    // Act
+    auto nodal_stresses = ExtrapolationUtilities::CalculateNodalStresses(
+        node_ids, element.GetGeometry(), element.GetIntegrationMethod(), cauchy_stress_vectors,
+        element.Id());
+
+    // Assert
+    KRATOS_EXPECT_EQ(nodal_stresses.size(), node_ids.size());
+
+    std::vector<Vector> expected_nodal_stresses;
+    expected_nodal_stresses.emplace_back(cauchy_stress_vectors[0] - delta_stress);
+    expected_nodal_stresses.push_back(cauchy_stress_vectors[1]);
+    expected_nodal_stresses.emplace_back(cauchy_stress_vectors[2] + delta_stress);
+
+    for (auto i = std::size_t{0}; i < nodal_stresses.size(); ++i) {
+        KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(nodal_stresses[i].value(), expected_nodal_stresses[i],
+                                           Defaults::relative_tolerance);
+    }
+
+    // a reduced number of node ids
+    node_ids.pop_back();
+    nodal_stresses = ExtrapolationUtilities::CalculateNodalStresses(node_ids, element.GetGeometry(),
+                                                                    element.GetIntegrationMethod(),
+                                                                    cauchy_stress_vectors, element.Id());
+    KRATOS_EXPECT_EQ(nodal_stresses.size(), node_ids.size());
+    for (auto i = std::size_t{0}; i < nodal_stresses.size(); ++i) {
+        KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(nodal_stresses[i].value(), expected_nodal_stresses[i],
+                                           Defaults::relative_tolerance);
+    }
+
+    // a wrong node id
+    node_ids       = {1, 4, 3};
+    nodal_stresses = ExtrapolationUtilities::CalculateNodalStresses(node_ids, element.GetGeometry(),
+                                                                    element.GetIntegrationMethod(),
+                                                                    cauchy_stress_vectors, element.Id());
+    KRATOS_EXPECT_EQ(nodal_stresses.size(), node_ids.size());
+    KRATOS_EXPECT_TRUE(nodal_stresses[1].has_value() == false)
```

Comment:
Using `KRATOS_EXPECT_FALSE` may be clearer here:
```suggestion
    KRATOS_EXPECT_FALSE(nodal_stresses[1].has_value())
```

Reply:
better ;)